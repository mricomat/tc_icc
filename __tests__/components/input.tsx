import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import Input from 'src/components/input';

describe('test Input', () => {
  test('it renders correctly', () => {
    const input = render(<Input />);
    expect(input).toMatchSnapshot();
  });

  test('it renders correctly with placeholder', () => {
    const { getByPlaceholderText } = render(<Input placeholder={'test'} />);
    const input = getByPlaceholderText('test');
    expect(input).not.toBe(null);
  });

  test('it shows correctly value text', () => {
    const { getByTestId } = render(<Input value={'testing'} />);
    const input = getByTestId('Input');
    expect(input).not.toBe(null);
    expect(input.props.value).toBe('testing');
  });

  test('it works onChangeText correctly', () => {
    const onChangeText = jest.fn();
    const { getByTestId } = render(<Input onChangeText={onChangeText} />);
    fireEvent.changeText(getByTestId('Input'), 'test');
    expect(onChangeText).toHaveBeenCalledWith('test');
  });
});
