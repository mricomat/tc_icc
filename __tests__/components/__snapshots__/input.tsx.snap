// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Input it renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {},
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    multiline={false}
    numberOfLines={1}
    onChangeText={[Function]}
    placeholderTextColor="#7F7F7F"
    rejectResponderTermination={true}
    selectionColor="#000000"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 6,
          "color": "#333333",
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        },
        undefined,
      ]
    }
    testID="Input"
    underlineColorAndroid="transparent"
  />
</View>
`;
